openmmtools/__init__.py

```diff
@@ -8,6 +8,3 @@
 # Define global version.
 from openmmtools import version
 __version__ = version.version
-
-# Import modules.
-from openmmtools import testsystems, integrators, alchemy, mcmc, states, cache, utils, constants, forces, forcefactories, storage, multistate
```

Comment:
I still get confused by Python 3 import system, but I'm not sure this is harmless as it may cause this to fail in some cases
```python
import openmmtools
openmmtools.integrators.LangevinIntegrator  # raises exception
```
which, if it actually happens, would break a bunch of other packages. Alternatively, if the goal is to fix #406, we can move the warning inside the `MultiStateSampler`, `MultiStateReporter`, and `MultiStateAnalyzer` class as
```python
class MultiStateSampler:
    warning.warn(message, stacklevel=2)
```
I _think_ the warning would be raised also if you try to use a class that inherit from those, although I didn't try it.

Reply:
Or maybe it would make sense to print that `warn` in the `__new__` magic method to make it support inheritance.

Reply:
Or also, we could selectively not import only `multistate`, or remove [these imports instead](https://github.com/choderalab/openmmtools/blob/master/openmmtools/multistate/__init__.py#L64-L70) after moving the warning inside the `mulstistate` submodules.

Reply:
@andrrizzi : This seems to work just fine:
```
>>> import openmmtools
/Users/choderaj/github/choderalab/openmmtools/openmmtools/openmmtools/multistate/__init__.py:77: UserWarning: Warning: openmmtools.multistate API is experimental
  warnings.warn('Warning: openmmtools.multistate API is experimental')
>>> openmmtools.integrators.LangevinIntegrator()
<openmmtools.integrators.LangevinIntegrator; proxy of <Swig Object of type 'OpenMM::CustomIntegrator *' at 0x109bc4900> >
```
Does it really raise an exception for you?

Reply:
Ah, I see!

Reply:
> Or maybe it would make sense to raise that warn in the __new__

I tried that, but couldn't make it work.

> Or also, we could selectively not import only multistate

I think that would just make things even more confusing.

> remove these imports instead after moving the warning inside the mulstistate submodules.

We definitely don't want to do that, since the subpackage organization into files is arbitrary and we want to mask that.

Reply:
Let's try it out then! It could be possible that it changed sometime in one of the new Python 3 versions.

Reply:
Ah, sorry, just seen that you added it to `__init__`. +1

Reply:
Do we use the idiom 
```
import openmmtools
openmmtools.subpackage.class()
```
a lot? It seems like we shouldn't _expect_ this to work.

Reply:
I don't think so, but I think we do use `from openmmtools.subpackage import class` a lot instead of `from openmmtools.subpackage.module import class`.